ambry-api/src/main/java/com/github/ambry/server/ServerSecurityService.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.server;
+
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.router.Callback;
+import com.github.ambry.router.FutureResult;
+import java.io.Closeable;
+import io.netty.channel.ChannelHandlerContext;
+import java.util.concurrent.Future;
+
+
+/**
+ * Responsible for performing any security validations on the HTTP2 connection terminating on server. Validations
+ * could involve authentication, authorization, security checks and so on which the implementation can decide.
+ * This could also involve setting headers while responding, based on the request.
+ * Exceptions are returned via {@link Callback}s on any validation failure.
+ */
+public interface ServerSecurityService extends Closeable {
```

Comment:
Move to `com.github.ambry.rest`? 
This can also help ambry-api get rid of netty dependency.